sale_invoice_delivery_state/models/sale.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("delivery_state")
+    def _get_invoice_status(self):
+        super()._get_invoice_status()
+        open_orders = self.filtered(
+            lambda s: s.invoice_status == "to invoice"
+            and s.partner_id.invoice_policy == "fully"
+            and s.delivery_state != "done"
+        )
+        open_orders.write({"invoice_status": "no"})
```

Comment:
Do not use write in computed field, this can generate weird behavior.
You can use a "update", (the update will loop on each record to assign the value)

As the filtered is already doing a loop, I will recommends to  just do a loop with an if

```suggestion
        for record in self:
            if (record.invoice_status == "to invoice"
                 and record.partner_id.invoice_policy == "fully"
                 and record.delivery_state != "done"
                 ):
            record.invoice_status = "no"
```

Reply:
@sebastienbeau , it's fixed.